lens-cube/src/main/java/org/apache/lens/cube/parse/StorageTableResolver.java

```diff
@@ -245,7 +246,105 @@ private void resolveDimStorageTablesAndPartitions(CubeQueryContext cubeql) throw
     }
   }
 
-  // Resolves all the storage table names, which are valid for each updatePeriod
+  /**
+   * Following storages are removed:
+   * 1. The storage is not supported by driver.
+   * 2. The storage is not in the valid storage list.
+   * 3. The storage is not in any time range in the query.
+   * 4. The storage having no valid update period.
+   *
+   * This method also creates a list of valid update periods and stores them into {@link StorageCandidate}.
+   *
+   * @param cubeql
+   */
+  private void resolveStorageTable(CubeQueryContext cubeql) throws LensException {
+    Set<Candidate> storageCandidates = cubeql.getCandidates();
+    Iterator<Candidate> it = storageCandidates.iterator();
+    while (it.hasNext()) {
+      Candidate c = it.next();
+      assert (c instanceof StorageCandidate);
+      StorageCandidate sc = (StorageCandidate) c;
+      String storageTable = sc.getStorageName();
+      if (!isStorageSupported(storageTable)) {
+        log.info("Skipping storage: {} as it is not supported", storageTable);
+        cubeql.addStoragePruningMessage(storageTable, new SkipStorageCause(SkipStorageCode.UNSUPPORTED));
+        it.remove();
+        continue;
+      }
+      String str = conf.get(CubeQueryConfUtil.getValidStorageTablesKey(sc.getFact().getName()));
+      List<String> validFactStorageTables = StringUtils.isBlank(str)
+                                            ? null
+                                            : Arrays.asList(StringUtils.split(str.toLowerCase(), ","));
+      // Check if storageTable is in the list of valid storages.
+      if (validFactStorageTables != null && !validFactStorageTables.contains(storageTable)) {
+        log.info("Skipping storage table {} as it is not valid", storageTable);
+        cubeql.addStoragePruningMessage(storageTable, new SkipStorageCause(SkipStorageCode.INVALID));
+        it.remove();
+        continue;
+      }
+
+      boolean remove = true;
+      SkipStorageCause cause = null;
+      for (TimeRange range : cubeql.getTimeRanges()) {
+        if (!client.isStorageTableCandidateForRange(storageTable, range.getFromDate(), range.getToDate())) {
+          continue;
+        } else if (!client.partColExists(storageTable, range.getPartitionColumn())) {
```

Comment:
Shouldn't this be checked for all storages ? 
Logic for this part looks wrong .. 

I feel for each Storage we should check
1. partition column . If false,  remove, break;
2. For each Time Range check isStorageTableCandidateForRange. Even if one isStorageTableCandidateForRange is true, it should be ok to continue.